### PR TITLE
Reflection probes best practices

### PR DESCRIPTION
**Pull request by Sanakan8472**
Fixes ezEngine/ezEngine#767

### CODE REVIEW

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 1]
```diff
@@ -11,24 +11,26 @@ Probes take the parent game object's scale into account. Thus, sphere probes can
...
+The captured reflection is only visible in the probe's influence volume. For sphere probes this is limited by their radius and for box probes by their extents.
 
 At runtime, all probes are sorted by their volume and the smallest probe is sampled first. If the reflection is transparent in the cubemap, then next bigger cubemap that influences the geometry is sampled next. This continues until we hit an opaque pixel in a cubemap or if the fallback reflection of the [sky light component](sky-light-component.md) is reached.
 
-The probe's `ReflectionProbeMode` by default is set to `static`, in which case the captured cubemap reflection is only updated once at the start or whenever the settings change. Alternatively it can be set to `dynamic`, in which case probe is updated  continuously.
+The probe's `ReflectionProbeMode` by default is set to `static`, in which case the captured cubemap reflection is only updated once at the start or whenever the settings change. Alternatively it can be set to `dynamic`, in which case probe is updated continuously.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:18]: ... the Probe is updated...

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 2]
```diff
@@ -11,24 +11,26 @@ Probes take the parent game object's scale into account. Thus, sphere probes can
...
 
 ## Component Properties
 
 Sphere reflection component:
 
 * `Radius`: The influence range of the probe. The reflection is only visible on geometry intersecting this radius.
 * `Falloff`: Percentage of the radius that is smoothly blended into other probes.
+* `SphereProjection`: Enables paralax correction to project the reflection onto the shape of the sphere's radius.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:26]: Remove the radius at the end. Last word I mean.

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 3]
```diff
@@ -11,24 +11,26 @@ Probes take the parent game object's scale into account. Thus, sphere probes can
...
 
 Box reflection component:
 
 * `Extents`: The extents of the box projection. The cubemap will be projected to this box.
 * `InfluenceScale`, `InfluenceShift`: The influence volume can be smaller than the projected volume (extents) of the probe. This can be useful if you have e.g. a long corridor that you want to place multiple probes in. Each will have the same projection but a different part of the projection volume will be set as the influence volume centred around a probe capture offset.
 * `PositiveFalloff`, `NegativeFalloff`: Percentage of the influence volume in each direction that is smoothly blended into other probes. The falloff is defined for each face of the box.
+* `BoxProjection`: Enables paralax correction to project the reflection onto the shape of the box's extents.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:33]: Also remove the extent here. Just shape of the box.

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 4]
```diff
@@ -38,6 +40,51 @@ Common properties for both sphere and box reflection probes that describe how th
 * `CaptureOffset`: The capture offset allows for the capture position of the probe to be decoupled from the game object position.
 * `ShowDebugInfo`: If enabled, a sphere with a preview of the probe cubemap is rendered at the position of the capture offset. Use this to check whether all desired objects contribute to the probe. Above the sphere will be a stack of other spheres that showcase the reflection with increased roughness.
 
+## Best Practices
+
+### Don't try to create mirrors with reflection probes
+
+Avoid mirrors or other perfectly reflective surfaces. Reflection probes are approximite and should not be used to emulate mirrors and other perfect reflectors.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:47]: Typo in approximate.

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 5]
```diff
@@ -38,6 +40,51 @@ Common properties for both sphere and box reflection probes that describe how th
...
+
+Even when using projection, the illusion of a perfect mirror immediately brakes down once objects are in the box-shaped room as the will be splatted in the box extends as seen here:
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:49]: ... as theY will be...

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 6]
```diff
@@ -38,6 +40,51 @@ Common properties for both sphere and box reflection probes that describe how th
...
+![Avoid perfect reflective surfaces](media/reflection-avoid-mirrors.jpg)
+
+### Ensure box projections have roughly the same length on each side
+
+If a box projection has very different dimensions on each axis you will run into stretching artifacts at the long ends of the box as can be seen here:
+
+![Box projection stretching](media/reflection-stretching.jpg)
+
+
+Instead of just one box probe, use multiple boxes to span long corridor to prevent this.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:60]: corridorS

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 7]
```diff
@@ -38,6 +40,51 @@ Common properties for both sphere and box reflection probes that describe how th
...
+To do this, take the existing box that spans the entire area and do the following:
+1. Changed it's Influence scale to e.g. 0.4 on the long axis, X in this case.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:62]: Change.  Not changed.

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 8]
```diff
@@ -38,6 +40,51 @@ Common properties for both sphere and box reflection probes that describe how th
...
+4. Duplicate the object in-place again.
+5. Set the new duplicate's influence shift of the X-axis to -1.
+
+This should result in the following image. The first box is highlighted to show the yellow influence box.
+
+![Box projection stretching fixed](media/reflection-stretching-fixed.jpg)
+
+### Don't align box probes perfectly with walls
+
+When usung `BoxProjection`, it is best to not try to align a box extents perfectly with a room. Projections are intended to anchor reflections, not as a mirror substitute.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:74]: a box's extent

File: pages/docs/graphics/lighting/reflection-probe-components.md [hunk 9]
```diff
@@ -38,6 +40,51 @@ Common properties for both sphere and box reflection probes that describe how th
...
+While the floor reflection now fits perfectly, there is no reflection on the wall mirror. This is because there is a door to another room and to avoid hard interpolation discontinuties between reflection probes, the falloff in that direction needs to be set. As the falloff moves inwards the mirror on the wall basically gets no contribution of the room's probe anymore.
+On the left wall this problem can be avoided by setting the falloff to zero but that is generally not an option in most cases.
+
+![Box reflection probe placement 1](media/reflection-box-placement1.jpg)
+
+Instead, increase the extents and set a falloff in each direction to fade into the neighboring reflection probes.
+
+![Box reflection probe placement 2](media/reflection-box-placement2.jpg)
+
+Once the floor material is replaced with a more reasonable one that is not a perfect mirror the missalignment will no longer be aparent while still provinding the desired parallax effect to the reflections.
```
Comment (jankrassnigg) [inline pages/docs/graphics/lighting/reflection-probe-components.md:86]: Typo in providing

**Review by jankrassnigg — APPROVED** [review]

File: pages/docs/graphics/lighting/reflection-probe-components.md (hunk 1 above)
Comment (Sanakan8472) [reply pages/docs/graphics/lighting/reflection-probe-components.md:18]: Thanks for the review, due to github not showing any updates I had to move the fixes to a separate PR: https://github.com/ezEngine/docs-src/pull/6

**Pull request merged**